cmd/frontend/graphqlbackend/code_hosts_test.go

```diff
@@ -0,0 +1,290 @@
+			APIRateLimitIntervalSeconds: pointers.Ptr(int32(1)),
+			GitRateLimitIntervalSeconds: pointers.Ptr(int32(1)),
+			GitRateLimitQuota:           pointers.Ptr(int32(1)),
+		},
+		{
+			ID:                          2,
+			URL:                         "gitlab.com",
+			Kind:                        extsvc.KindGitLab,
+			APIRateLimitQuota:           pointers.Ptr(int32(2)),
+			APIRateLimitIntervalSeconds: pointers.Ptr(int32(2)),
+			GitRateLimitIntervalSeconds: pointers.Ptr(int32(2)),
+			GitRateLimitQuota:           pointers.Ptr(int32(2)),
+		},
+		{
+			ID:   3,
+			URL:  "bitbucket-cloud.com",
+			Kind: extsvc.KindBitbucketCloud,
+		},
+		{
+			ID:                          4,
+			URL:                         "bitbucket.com",
+			Kind:                        extsvc.KindBitbucketServer,
+			APIRateLimitQuota:           pointers.Ptr(int32(4)),
+			APIRateLimitIntervalSeconds: pointers.Ptr(int32(4)),
+			GitRateLimitIntervalSeconds: pointers.Ptr(int32(4)),
+			GitRateLimitQuota:           pointers.Ptr(int32(4)),
+		},
+	}
+
+	tests := []struct {
+		first int
+		after int32
+	}{
+		{
+			first: 1,
+			after: 0,
+		},
+		{
+			first: 1,
+			after: 1,
+		},
+		{
+			first: 1,
+			after: 2,
+		},
+		{
+			first: 1,
+			after: 3,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("first=%d after=%d", tc.first, tc.after), func(t *testing.T) {
+			store := dbmocks.NewMockCodeHostStore()
+			store.CountFunc.SetDefaultReturn(4, nil)
+			testCodeHost := testCodeHosts[tc.after]
+
+			store.ListFunc.SetDefaultHook(func(ctx context.Context, opts database.ListCodeHostsOpts) ([]*types.CodeHost, int32, error) {
+				assert.Equal(t, tc.first, opts.Limit)
+				assert.Equal(t, tc.after, opts.Cursor)
+				next := tc.after + int32(tc.first)
+				if int(next) >= len(testCodeHosts) {
+					next = 0
+				}
+
+				return testCodeHosts[tc.after : tc.after+int32(tc.first)], next, nil
+			})
+			users := dbmocks.NewMockUserStore()
+			users.GetByCurrentAuthUserFunc.SetDefaultReturn(&types.User{SiteAdmin: true}, nil)
+
+			eSvcs := []*types.ExternalService{
+				{ID: 1, DisplayName: "GITLAB #1"},
+				{ID: 2, DisplayName: "GITLAB #2"},
+			}
+			externalServices := dbmocks.NewMockExternalServiceStore()
+			externalServices.ListFunc.SetDefaultHook(func(ctx context.Context, options database.ExternalServicesListOptions) ([]*types.ExternalService, error) {
+				assert.Equal(t, options.CodeHostID, testCodeHost.ID)
+				assert.Equal(t, options.Limit, tc.first)
+				assert.Equal(t, options.Offset, 0)
+				return eSvcs, nil
+			})
+
+			ctx := context.Background()
+			db := dbmocks.NewMockDB()
+			db.CodeHostsFunc.SetDefaultReturn(store)
+			db.UsersFunc.SetDefaultReturn(users)
+			db.ExternalServicesFunc.SetDefaultReturn(externalServices)
+			variables := map[string]any{
+				"first": tc.first,
+			}
+
+			gqlAfterID := MarshalCodeHostID(tc.after)
+			if tc.after != 0 {
+				variables["after"] = gqlAfterID
+			}
+			var wantEndCursor *string
+			hasNext := false
```

Comment:
```suggestion
			wantHasNext := false
```

No? It's just like the value next to it, the `endCursor`